lib/classes/Service.js

```diff
@@ -149,16 +149,27 @@ class Service {
         // setup function.name property
         const stageNameForFunction = options.stage || this.provider.stage;
         _.forEach(that.functions, (functionObj, functionName) => {
+          const targetFunction = that.functions[functionName];
           if (!functionObj.events) {
-            that.functions[functionName].events = [];
+            targetFunction.events = [];
           }
           if (!_.isArray(functionObj.events)) {
             throw new SError(`Events for "${functionName}" must be an array,` +
                              ` not an ${typeof functionObj.events}`);
           }
 
+          // Translate HTTP events written in string notation to object notation
+          _.forEach(functionObj.events, (event, index) => {
```

Comment:
Is there a way that we move this from the `Service` class to the AWS API Gateway plugin?

The reason for this is that the Serverless core should be as provider independent as possible.